build: update @electron/docs-parser and @electron/typescript-definitions

devm33 opened the pull request:
#### Description of Change

Upgrades docs-parser and typescript-definitions to add new 'electron/utility' namespace added in:
- https://github.com/electron/docs-parser/pull/95
- https://github.com/electron/typescript-definitions/pull/246
- https://github.com/electron/typescript-definitions/pull/247

Also removes obsolete remote interface: https://github.com/electron/typescript-definitions/pull/249

<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Contributors guide: https://github.com/electron/electron/blob/main/CONTRIBUTING.md
-->

#### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: no-notes
<!-- Please add a one-line description for app developers to read in the release notes, or 'none' if no notes relevant to app developers. Examples and help on special cases: https://github.com/electron/clerk/blob/main/README.md#examples -->

jkleinsc reviewed (changes requested):
Due to security restrictions, yarn.lock changes are restricted to maintainers only.

Additionally, it appears that this change does not pass lint: https://app.circleci.com/pipelines/github/electron/electron/75605/workflows/dcf582a9-b884-415b-8c2f-56bf047d13d0/jobs/1622825

devm33 commented:
Thanks for taking a look!

> Due to security restrictions, yarn.lock changes are restricted to maintainers only.

Should the PR only include package.json changes or does a maintainer need to do this PR?

> Additionally, it appears that this change does not pass lint: https://app.circleci.com/pipelines/github/electron/electron/75605/workflows/dcf582a9-b884-415b-8c2f-56bf047d13d0/jobs/1622825

Sorry yes working on fixing! https://github.com/electron/electron/pull/40017#issuecomment-1769813375

devm33 commented:
@jkleinsc I've updated here with the fix for the typescript-definitions change https://github.com/electron/typescript-definitions/pull/247 and dropped the yarn.lock changes. Unfortunately, that does mean lint is now failing due to the package.json and yarn.lock being out of sync.

``error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`.`` https://app.circleci.com/pipelines/github/electron/electron/75794/workflows/7803b286-c927-4c3f-b3a2-48cfa187f9d5/jobs/1626190

dsanders11 commented:
I've added the `yarn.lock` changes myself and pushed them, but it should be noted this lands multiple changes from these packages (including dropping the remote interface stuff), so I think the title should probably be more generic so it better matches all of the functional changes, e.g. "build: update @electron/docs-parser and @electron/typescript-definitions".

FYI @devm33, I think you may be using an older version of `yarn` which is causing your `yarn.lock` changes to be bigger than necessary. Your original `yarn.lock` changes were in the thousands of lines, while mine is less than 10.

devm33 commented:
Thanks @dsanders11! Updated the PR title and description
